sdk/communication/Shared/src/CommunicationIdentifierSerializer.cs

```diff
@@ -0,0 +1,86 @@
+    {
+        private const string OrgIdPrefix = "8:orgid:";
+        private const string TeamsVisitorPrefix = "8:teamsvisitor:";
+
+        public static CommunicationIdentifier Deserialize(CommunicationIdentifierModel identifier)
+        {
+            var id = identifier.Id;
+            var kind = identifier.Kind;
+            return kind switch
+            {
+                IdentifierKinds.CommunicationUser
+                    => new CommunicationUserIdentifier(AssertNotNull(id, nameof(identifier.Id), kind)),
+                IdentifierKinds.CallingApplication
+                    => new CallingApplicationIdentifier(AssertNotNull(id, nameof(identifier.Id), kind)),
+                IdentifierKinds.PhoneNumber
+                    => new PhoneNumberIdentifier(AssertNotNull(identifier.PhoneNumber, nameof(identifier.PhoneNumber), kind)),
+                IdentifierKinds.MicrosoftTeamsUser
+                    => new MicrosoftTeamsUserIdentifier(
+                        GetTeamsUserId(AssertNotNull(identifier.MicrosoftTeamsUserId, nameof(identifier.MicrosoftTeamsUserId), kind)),
+                        AssertNotNull(identifier.IsAnonymous, nameof(identifier.IsAnonymous), kind)),
+                IdentifierKinds.Unknown
+                    => new UnknownIdentifier(AssertNotNull(id, nameof(identifier.Id), kind)),
+                _
+                    => new UnknownIdentifier(AssertNotNull(id, nameof(identifier.Id), kind)),
+            };
+
+            static string GetTeamsUserId(string fullId)
+            {
+                if (fullId.StartsWith(OrgIdPrefix, StringComparison.InvariantCulture))
+                    return fullId.Substring(OrgIdPrefix.Length);
+                if (fullId.StartsWith(TeamsVisitorPrefix, StringComparison.InvariantCulture))
+                    return fullId.Substring(TeamsVisitorPrefix.Length);
+                throw new JsonException("Invalid MicrosoftTeams user identifier string");
+            }
+        }
+
+        public static CommunicationIdentifierModel Serialize(CommunicationIdentifier identifier)
+            => identifier switch
+            {
+                CommunicationUserIdentifier u => new CommunicationIdentifierModel(IdentifierKinds.CommunicationUser) { Id = u.Id },
+                CallingApplicationIdentifier a => new CommunicationIdentifierModel(IdentifierKinds.CallingApplication) { Id = a.Id },
+                PhoneNumberIdentifier p => new CommunicationIdentifierModel(IdentifierKinds.PhoneNumber) { PhoneNumber = p.Value },
+                MicrosoftTeamsUserIdentifier u => new CommunicationIdentifierModel(IdentifierKinds.MicrosoftTeamsUser) { MicrosoftTeamsUserId = GetFullTeamsUserId(u.UserId, u.IsAnonymous) },
+                UnknownIdentifier u => new CommunicationIdentifierModel(IdentifierKinds.Unknown) { Id = u.Id },
+                _ => throw new NotSupportedException(),
+            };
+
+        private static string GetFullTeamsUserId(string userId, bool isAnonymous)
```

Comment:
We can get rid of this now because we require microsoftTeamsUserId to be set for Teams user when serialized